app/controllers/feature_controller.rb

```diff
@@ -0,0 +1,45 @@
+class FeatureController < ApplicationController
+  before_action :validate_feature!
+
+  # Toggles boolean for session based feature flags
+  def toggle
```

Comment:
A general ticket is cool so I don't forget but don't want to hold off on the merge if that's cool.